### DISCUSSION TITLE
-P 5004:6004 vs -P 6004

### DISCUSSION THREAD

**Discussion by alatteri** [CESNET/UltraGrid#195] [Q&A] [answered]
Hi,

In the recent update to the Reflector documentation, you suggest using -P 5004:6004:5006:6006 to remap ports on the UG command.  I am wondering, what is the advantage or difference of using -P 5004:6004 versus only stating -P 6004??

`hostA $ uv -t testcard -d gl -s testcard -r portaudio -P 5004:6004:5006:6006 localhost
`
I'm not clear on what having two ports separated by the colon does, versus declaring only the desired port?

Thanks.

**Comment by alatteri** [comment, marked as answer]
I think I understand.  In the situation where you are using UG as both sender and receiver on the same machine.

5004            :       6004
^^^^^^^^^   ^^^^^^^^^
Sender Port : Reciever Port

**Reply by MartinPulec** [reply]
Yes, exactly as you say - I've also added your remark to the doc explicitly (semantics of individual numbers). Of course it would be also possible to run separated sender and receiver:

    uv -s testcard -t testcard -P 6004 localhost
    uv -d sdl -r portaudio <remote_peer>

which has a simplier syntax - UG does a little tweaking with ports, so if only one value is given, it is meant as a _base_ port which then affects all other ports. With the colon-separated syntax you are saying that you want to explicitly set all ports.
